mp4parse/src/lib.rs

```diff
@@ -1155,11 +1155,13 @@ fn read_ctts<T: Read>(src: &mut BMFFBox<T>) -> Result<CompositionOffsetBox> {
     let (version, _) = read_fullbox_extra(src)?;
 
     let counts = be_u32(src)?;
+
+    if src.bytes_left() < (counts as usize * std::mem::size_of::<TimeOffsetVersion>()) {
```

Comment:
Thanks for hoisting this. I think `(counts as usize * 8)` would be better here. One could add new fields to `TimeOffsetVersion` and change this calculation without changing the number of bytes we're reading.